frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/ShowSelectedBranchInGitLogAction.java

```diff
@@ -0,0 +1,107 @@
+package com.virtuslab.gitmachete.frontend.actions.contextmenu;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.format;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.util.Objects;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.vcs.log.Hash;
+import com.intellij.vcs.log.impl.VcsLogContentUtil;
+import com.intellij.vcs.log.impl.VcsProjectLog;
+import git4idea.branch.GitBranchesCollection;
+import lombok.CustomLog;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.frontend.actions.base.BaseGitMacheteRepositoryReadyAction;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
+import com.virtuslab.logger.IEnhancedLambdaLogger;
+
+@CustomLog
+public class ShowSelectedBranchInGitLogAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeySelectedBranchName {
+
+  @Override
+  public IEnhancedLambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    val selectedBranchName = getSelectedBranchName(anActionEvent);
+    // It's very unlikely that selectedBranchName is empty at this point since it's assigned directly before invoking this
+    // action in GitMacheteGraphTable.GitMacheteGraphTableMouseAdapter.mouseClicked; still, it's better to be safe.
+    if (selectedBranchName.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getString("action.GitMachete.ShowSelectedBranchInGitLogAction.undefined.branch-name"));
+      return;
+    }
+
+    presentation.setDescription(
+        format(getString("action.GitMachete.ShowSelectedBranchInGitLogAction.description.precise"), selectedBranchName.get()));
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    val selectedBranchName = getSelectedBranchName(anActionEvent);
+    if (selectedBranchName.isEmpty()) {
+      return;
+    }
+
+    val project = getProject(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+
+    if (gitRepository.isDefined()) {
+      log().debug(() -> "Queuing show '${selectedBranchName.get()}' branch in Git log background task");
+
+      GitBranchesCollection branches = gitRepository.get().getBranches();
+      @SuppressWarnings({"nullness:argument", "nullness:return"}) val maybeHash = selectedBranchName
```

Comment:
🤔 

What are the Checker errors raised here?

Reply:
```java
/home/mikko/IdeaProjects/git-machete-intellij-plugin/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/ShowSelectedBranchInGitLogAction.java:75: error: [[fbc, nullness, allcheckers]:methodref.return] Incompatible return type
          .map(branches::findBranchByName).filter(Objects::nonNull).map(branches::getHash);
               ^
  found   : @Initialized @Nullable GitBranch
  required: @Initialized @NonNull GitBranch
  Consequence: method in @Initialized @NonNull GitBranchesCollection
    @Initialized @Nullable GitBranch findBranchByName(@Initialized @NonNull GitBranchesCollection this, @Initialized @NonNull String p0)
  is not a valid method reference for method in @Initialized @NonNull Function<@Initialized @NonNull String, @Initialized @NonNull GitBranch>
    @Initialized @NonNull GitBranch apply(@Initialized @NonNull Function<@Initialized @NonNull String, @Initialized @NonNull GitBranch> this, @Initialized @NonNull String p0)
/home/mikko/IdeaProjects/git-machete-intellij-plugin/frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/contextmenu/ShowSelectedBranchInGitLogAction.java:75: error: [[fbc, nullness, allcheckers]:methodref.return] Incompatible return type
          .map(branches::findBranchByName).filter(Objects::nonNull).map(branches::getHash);
                                                                        ^
  found   : @Initialized @Nullable Hash
  required: @Initialized @NonNull Hash
  Consequence: method in @Initialized @NonNull GitBranchesCollection
    @Initialized @Nullable Hash getHash(@Initialized @NonNull GitBranchesCollection this, @Initialized @NonNull GitBranch p0)
  is not a valid method reference for method in @Initialized @NonNull Function<@Initialized @NonNull GitBranch, @Initialized @NonNull Hash>
    @Initialized @NonNull Hash apply(@Initialized @NonNull Function<@Initialized @NonNull GitBranch, @Initialized @NonNull Hash> this, @Initialized @NonNull GitBranch p0)
```

Reply:
Now I see that "nullness:argument" is not needed, removing

Reply:
Okay, makes sense